app/src/test/java/com/concordium/wallet/data/util/PreferencesMigrationShould.kt

```diff
@@ -0,0 +1,61 @@
+package com.concordium.wallet.data.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.test.core.app.ApplicationProvider
+import com.concordium.wallet.data.preferences.AuthPreferences
+import com.concordium.wallet.data.preferences.Preferences
+import com.concordium.wallet.data.preferences.SharedPreferencesKeys
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+
+/**
+ * Unit Tests for the migration from unencrypted to encrypted [SharedPreferences] performed in [Preferences]
+ */
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class PreferencesMigrationShould {
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+
+    @Test
+    fun encryptAndClearOldPreferences(){
+        val unencryptedPreferences = context.getSharedPreferences(SharedPreferencesKeys.PREF_FILE_AUTH, Context.MODE_PRIVATE)
```

Comment:
Since you're using `AuthPreferences` it's not clear that you're actually testing it's superclass `Preferences`. At least I don't really see what `PREFKEY_HAS_SETUP_USER` and `PREFKEY_BIOMETRIC_KEY` has to do with encrypted prefs.

Is there a reason that you cannot use that class directly? Or make a small test implementation. Or am I missing something?

Reply:
`PREFKEY_HAS_SETUP_USER` and `PREFKEY_BIOMETRIC_KEY` are used for the test, they are not important. 
We set them first in the unencrypted SharedPreferences and then after we initialise the AuthPreference we check to see if they are migrated to the encrypted SharedPreferences. 
`authPreferences.getHasSetupUser()` returns the value for `PREFKEY_HAS_SETUP_USER`
`authPreferences.getAuthKeyName()` return the value for `PREFKEY_BIOMETRIC_KEY`
also we check if the old unencrypted SharedPreferences are empty
The functions in Preference are all private and can't be used directly, so I think testing AuthPreferences is enough to showcase that the migration to the encrypted SharedPreference works as intended.